lib/config/decrypt.ts

```diff
@@ -1,11 +1,41 @@
 import crypto from 'crypto';
 import is from '@sindresorhus/is';
+import * as openpgp from 'openpgp';
 import { logger } from '../logger';
 import { maskToken } from '../util/mask';
 import { add } from '../util/sanitize';
 import { getGlobalConfig } from './global';
 import type { RenovateConfig } from './types';
 
+export async function tryDecryptPgp(
+  privateKey: string,
+  encryptedStr: string
+): Promise<string | null> {
+  if (encryptedStr.length < 500) {
+    // optimization during transition of public key -> pgp
+    return null;
+  }
+  try {
+    const pk = await openpgp.readPrivateKey({
+      // prettier-ignore
+      armoredKey: privateKey.replace(/\n[ \t]+/g, '\n'), // little massage to help a common problem
+    });
+    const armoredMessage = `-----BEGIN PGP MESSAGE-----\n\n${encryptedStr}\n-----END PGP MESSAGE-----`;
```

Comment:
If you self-encrypt you need to remove those pgp comments? Can't they are optional?

This should at lease be documented how to generate the secrets

Reply:
I might refactor it to make these optional